crossdock/client/client.go

```diff
@@ -36,19 +38,24 @@ func behaviorRequestHandler(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	// All behaviors will eventually contribute to this behavior's output.
-	bt := BehaviorTester{Params: httpParams{Request: r}}
-	v := bt.Param("behavior")
-	switch v {
-	case "raw", "json", "thrift":
-		runEchoBehavior(&bt, v)
-	default:
-		bt.NewBehavior(BasicEntryBuilder).Skipf("unknown behavior %v", v)
+	var s behavior.EntrySink
+	behavior.Run(func() { dispatch(&s, httpParams{r}) })
```

Comment:
one thing that would be nice -- if the function panics for some reason, instead of crashing the process, we should recover, and log the panic as a fatalf (this is what the test framework does with test functions).